pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1053,6 +1061,47 @@ func (p *PriorityQueue) moveAllToActiveOrBackoffQueue(logger klog.Logger, event
 	p.movePodsToActiveOrBackoffQueue(logger, unschedulablePods, event, oldObj, newObj)
 }
 
+// RunInBackground implements BackgroundRunner.
+func (p *PriorityQueue) RunInBackground(ctx context.Context, pod *v1.Pod, cb func(ctx context.Context) error) {
+	logger := klog.FromContext(ctx)
+	logger = klog.LoggerWithName(logger, "Background")
+	// TODO: can we force background activities to stop via
+	// some other context?
+	ctx = klog.NewContext(context.Background(), logger)
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				p.podFailure(ctx, pod, fmt.Errorf("background operation failed with a panic: %v", r))
+			}
+		}()
+		if err := cb(ctx); err != nil {
+			p.podFailure(ctx, pod, fmt.Errorf("background operation failed: %v", err))
+		}
+	}()
+}
+
+// podFailure triggers scheduling of the pod by emitting a special event.  If
+// that happens while the pod is in flight, re-enqueuing the pod will pick the
+// backoff queue. If it happens later, the pod gets moved out of the
+// unschedulable queue into the active or backoff queue, depending on whether
+// it has failed before.
+func (p *PriorityQueue) podFailure(ctx context.Context, pod *v1.Pod, err error) {
+	logger := klog.FromContext(ctx)
+	logger.V(3).Info("Scheduling the pod will be retried", "pod", klog.KObj(pod), "err", err)
+	p.MoveAllToActiveOrBackoffQueue(logger, podFailed, pod, pod, func(unschedulablePod *v1.Pod) bool {
+		return pod.UID == unschedulablePod.UID
```

Comment:
This is the opposite of https://github.com/kubernetes/kubernetes/blob/aa4ec3c5b006157751df4feac5c620c9dfadd2b7/pkg/scheduler/schedule_one.go#L341: only this pod is affected, not every other pod.